src/FuncSharp/Collections/IEnumerableExtensions_ReadOnlyList.cs

```diff
@@ -1,100 +1,122 @@
+        return collection is not null && collection.Count > 0;
     }
 
-    public static bool IsEmpty<T>(this IReadOnlyList<T> source)
+    [Pure]
+    public static bool IsEmpty<T>(this IReadOnlyCollection<T> collection)
     {
-        return !source.NonEmpty();
+        return !collection.NonEmpty();
     }
 
-    public static bool IsMultiple<T>(this IReadOnlyList<T> source)
+    [Pure]
+    public static bool IsMultiple<T>(this IReadOnlyCollection<T> collection)
     {
-        return source is not null && source.Count > 1;
+        return collection is not null && collection.Count > 1;
     }
 
-    public static bool IsSingle<T>(this IReadOnlyList<T> source)
+    [Pure]
+    public static bool IsSingle<T>(this IReadOnlyCollection<T> collection)
     {
-        return source is not null && source.Count == 1;
+        return collection is not null && collection.Count == 1;
     }
 
-    public static T Single<T>(this IReadOnlyList<T> source)
+    [Pure]
+    public static T Single<T>(this IReadOnlyList<T> list)
     {
-        if (!source.IsSingle())
-        {
-            throw new ArgumentException("Source is not a single element.");
-        }
-
-        return source[0];
+        return list.IsSingle()
+            ? list[0]
+            : throw new ArgumentException("Source is not a single element.");
     }
 
-    public static IOption<T> SingleOption<T>(this IReadOnlyList<T> source)
+    [Pure]
+    public static IOption<T> SingleOption<T>(this IReadOnlyList<T> list)
     {
-        return source.IsSingle()
-            ? Option.Valued(source[0])
+        return list.IsSingle()
+            ? Option.Valued(list[0])
             : Option.Empty<T>();
     }
 
-    public static T First<T>(this IReadOnlyList<T> source)
+    [Pure]
+    public static T First<T>(this IReadOnlyList<T> list)
     {
-        return source.IsEmpty()
-            ? throw new ArgumentException("Source is empty.")
-            : source[0];
+        return list.ElementAt(0);
     }
 
-    public static IOption<T> FirstOption<T>(this IReadOnlyList<T> source)
+    [Pure]
+    public static IOption<T> FirstOption<T>(this IReadOnlyList<T> list)
     {
-        return source.IsEmpty()
+        return list.IsEmpty()
             ? Option.Empty<T>()
-            : Option.Valued(source[0]);
+            : Option.Valued(list[0]);
+    }
+
+    [Pure]
+    public static T Second<T>(this IReadOnlyList<T> list)
+    {
+        return list[1];
+    }
+
+    public static T Third<T>(this IReadOnlyList<T> list)
+    {
+        return list[2];
+    }
+
+    public static T Fourth<T>(this IReadOnlyList<T> list)
+    {
+        return list[3];
     }
 
-    public static T Second<T>(this IReadOnlyList<T> source)
+    public static T Fifth<T>(this IReadOnlyList<T> list)
     {
-        return source.Count <= 1
-            ? throw new ArgumentException("Source has less than 2 elements.")
-            : source[1];
+        return list[4];
```

Comment:
This looks dangerous. I smell potential `IndexOutOfRangeException`. 

I guess we could scratch this and let callers use whatever they need and make it their responsibility.

Not to mention this could even throw a `NullReferenceException`.

If we want to make a safe method that throws our relevant exceptions, we could go with our implementation of `ElementAt` where the caller provides the list and index number 😊 

EDIT: oh nvm, we have ElementAt 😅

Reply:
We're gonna keep it for compatibility reasons + it's just a syntax sugar